request.go

```diff
@@ -0,0 +1,86 @@
+package katapult
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// Request represents a HTTP request to the Katapult API, it is essentially
+// similar to http.Request, but stripped down to the bare essentials, with some
+// Katapult-specific attributes added.
+type Request struct {
+	// Method is the HTTP method to perform when making the request.
+	Method string
+
+	// URL is the request URL to perform against Katapult's API. Generally the
+	// only fields you need to set are Path and RawQuery, as it will be merged
+	// with the Client's BaseURL value through its ResolveReference() method.
+	URL *url.URL
+
+	// NoAuth instructs Client not to send Authorization header containing the
+	// APIKey. This is useful for public endpoints which do not require/use
+	// authentication.
+	NoAuth bool
+
+	// Header holds request-specific HTTP headers. Client.Do() will set a number
+	// of essential headers itself which cannot be customized through
+	// Request.Headers.
+	Header http.Header
+
+	// ContentType allows sending a custom request body of any mimetype. If set
+	// Body must be a io.Reader. If not set Body must be a object which can be
+	// serialized with json.Marshal(). Content-Type header is only sent when
+	// Body is not nil.
+	ContentType string
+
+	// Body can be any object which can be marshaled to JSON through
+	// json.Marshal() when ContentType is not set. If ContentType is set, Body
+	// must be a io.Reader, or nil.
+	//
+	// No validation is performed between Method and Body, making it possible to
+	// send a body with a method that does not allow it.
+	Body interface{}
+}
+
+func NewRequest(method string, u *url.URL, body interface{}) *Request {
```

Comment:
I wonder if we can avoid a constructor here?

Reply:
It's not strictly needed, no. But I figured it might be a nice convenience method, and gives us "standard" way to create a Request, and gives us the flexibility to define a reasonable default set of fields for requests if we need expand the struct in the future.

That said, I'm not 100% sure what convention is around the existence of `NewFoo` functions, only create them when you explicitly need them, or create them when it "feels" right? lol

Reply:
I've always felt they are handy if you have a struct with lots of fields, but only some of them need to be set by a consumer of your API. Hence, it's a bit clearer to a user.

I think in this case, its pretty harmless for us to include.